Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,314 @@
+
+```qsharp
+operation UsingAndBorrowing() : Result {
+    use a = Qubit();
+    borrow b = Qubit();
+    // ...
+    return M(a); // a and b are released here.
+}
+```
+
+Example 5b: A qubit allocated within a new block and released before the end of the outer block.
+
+```qsharp
+operation DifferentLifetime() : Result {
+    mutable r = Zero;
+    use q = Qubit() {
+        // Apply operations...
+        set r = M(q);
+        // q is released here.
+    }
+    // Classical post-processing...
+    return r;
+}
+```
+
+# Implementation
+
+The `use` and `borrow` keywords can be added, and the `using` and `borrowing` keywords can be deprecated.
+Until `using` and `borrowing` are removed, they can be used in place of `use` and `borrow` for both the block and non-block statements.
+
+The block for the existing `using` and `borrowing` syntax nodes can be made optional.
+Either the block remains optional throughout all stages of the syntax tree, or the non-block form is transformed into the block form by automatically moving all statements below it into a new block.
+In the former case, all syntax tree consumers need to be aware of both forms, while in the latter case consumers would notice no change from the current syntax tree node.
+
+## Timeline
+
+There are no dependencies on other proposals or libraries.
+The work needed for either option appears to be minimal, since it is similar to the existing implementation.
+
+# Further Considerations
+
+Development tools should consider adding support for transitioning from the current syntax to the proposed syntax, such as a code action in IDEs.
+
+## Related Mechanisms
+
+This proposal relies on the existing qubit management functionality that is already a core feature of Q#.
+It does not change the behavior of qubit management; it only provides new syntax for existing functionality.
+
+## Impact on Existing Mechanisms
+
+The new keywords `use` and `borrow` were previously valid Q# identifiers.
+Adding them is a breaking change, unless an opt-in mechanism for new keywords is added to the language.
+
+For consistency, parentheses are no longer required for all block statement headers.
+That means that the following code is now valid:
+
+```qsharp
+if M(q) == One {
+    X(q);
+}
+
+for x in xs {
+    Message(x);
+}
+```
+
+## Anticipated Interactions with Future Modifications
+
+There will likely be changes to qubit initializers soon.
+See [Issue #40: Allocatable types and generalization of initializers](https://github.com/microsoft/qsharp-language/issues/40).
+We expect the changes to initializers to integrate well with this proposal, because this proposal only changes the syntax *around* qubit initializers, not the initializers themselves.
+
+Removing parentheses around block statement headers may affect future syntax development.
+For example, it would not be possible to make the braces optional without also re-introducing parentheses or another token, because of parsing issues with code like:
```

Comment:
The better alternative to mention in this case I believe would be to move to having a keyword taking the role of the delimiter, e.g. 
```
if M(q) == One then
    X(q);
```